src/pages/Homepage.tsx

```diff
@@ -26,27 +26,26 @@ const Homepage = (): ReactElement => {
       name: e.name,
       location: e.location,
       description: e.description,
-      color: getColor(new Date(e.startTime), new Date(e.endTime)),
+      status: getStatus(new Date(e.startTime), new Date(e.endTime)),
     };
   });
 
-  const liveEvents = events
-    .filter((e) => e.color === "bg-sga-red")
-    .map((e) => (
-      <>
-        <EventCard {...e} />
-        <hr className="border-black home-mx" />
-      </>
-    ));
+  const liveEvents: ReactElement[] = [];
+  const upcomingEvents: ReactElement[] = [];
 
-  const upcomingEvents = events
-    .filter((e) => e.color !== "bg-sga-red")
-    .map((e) => (
-      <>
-        <EventCard {...e} />
-        <hr className="border-black home-mx" />
-      </>
-    ));
+  // Filters events into their corresponding status.
+  events.reduce(
```

Comment:
I _think_ something like the below would work for conditionally adding a line divider if the events are on different dates (I'm assuming we'll never have dividers for live events since they'd be happening on the same dates? Unless there's an event that spans more than one day but it still makes sense not to have a divider when two events are live at the same time). Lmk what you think! I also haven't tested it cuz I'm too lazy to set up the branch on my computer lmaO

```suggestion
  events.reduce(
  function (result, curr, i, events) {
      if (curr.status === Status.Live) {
        result[0].push(<EventCard key={curr.name} {...curr} />);
      } else {
        // if this event is on a different day than the event before it, add a horizontal divider
        if (i > 0 && !datesOnSameDay(events[i], events[i - 1])) {
          result[0].push(<hr className="border-black home-mx" />)
        }
        result[1].push(<EventCard key={curr.name} {...curr} />);
      }

      return result;
    },
    [liveEvents, upcomingEvents]
  );
```

also helper function; I think this would also be defined in the Homepage or it could be defined and imported from the EventCard file

`
const datesOnSameDay = (date1, date2) =>
    date1.getFullYear() === date2.getFullYear() &&
    date1.getMonth() === date2.getMonth() &&
    date1.getDate() === date2.getDate();
`

Reply:
Thanks for pointing that out!! i will change